wait/host.go

```diff
@@ -105,45 +82,29 @@ func UntilHasUserAccountStatus(expUaStatuses ...toolchainv1alpha1.UserAccountSta
 	}
 }
 
-// WaitForUserSignupStatusConditions waits until there is a UserSignup available with the given name and set of status conditions
-func (a *HostAwaitility) WaitForUserSignupStatusConditions(name string, conditions ...toolchainv1alpha1.Condition) error {
-	return wait.Poll(RetryInterval, Timeout, func() (done bool, err error) {
-		userSignup := &toolchainv1alpha1.UserSignup{}
+// WaitForUserSignup waits until there is a UserSignup available with the given name and set of status conditions
+func (a *HostAwaitility) WaitForUserSignup(name string, conditions ...toolchainv1alpha1.Condition) (*toolchainv1alpha1.UserSignup, error) {
+	userSignup := &toolchainv1alpha1.UserSignup{}
+	err := wait.Poll(RetryInterval, Timeout, func() (done bool, err error) {
 		if err := a.Client.Get(context.TODO(), types.NamespacedName{Namespace: a.Ns, Name: name}, userSignup); err != nil {
 			if errors.IsNotFound(err) {
 				a.T.Logf("waiting for availability of UserSignup '%s'", name)
 				return false, nil
 			}
 			return false, err
 		}
-
 		if test.ConditionsMatch(userSignup.Status.Conditions, conditions...) {
```

Comment:
This can be a bit tricky now - in case the conditions are not specified (ie. we don't need to check them) this condition check could fail because the `ConditionMatch` checks also the count of the statuses: https://github.com/codeready-toolchain/toolchain-common/blob/master/pkg/test/condition.go#L41
I think that we should add:
* either "wait condition" similar to what we have for MUR: https://github.com/codeready-toolchain/toolchain-e2e/pull/31/files#diff-dfba07fbdde19dfb6e7d80277c03139eR51
* or an "if statement" that would skip this `ConditionMatch` in case no condition is provided

Reply:
ok, so I closed https://github.com/codeready-toolchain/toolchain-common/pull/51 and rather, I implemented the first option in 8032a7d